core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,106 +17,96 @@
 
 package kafka.utils
 
-import org.apache.log4j.Logger
+import com.typesafe.scalalogging.{LazyLogging, Logger}
+import org.slf4j.{Marker, MarkerFactory}
 
-trait Logging {
-  val loggerName = this.getClass.getName
-  lazy val logger = Logger.getLogger(loggerName)
+import scala.util.{Failure, Success, Try}
 
-  protected var logIdent: String = null
+object Log4jControllerRegistration {
+  private val logger = Logger(this.getClass.getName)
 
-  // Force initialization to register Log4jControllerMBean
-  private val log4jController = Log4jController
+  val FatalMarker: Marker = MarkerFactory.getMarker("FATAL")
 
-  protected def msgWithLogIdent(msg: String) =
-    if (logIdent == null) msg else logIdent + msg
-
-  def trace(msg: => String): Unit = {
-    if (logger.isTraceEnabled())
-      logger.trace(msgWithLogIdent(msg))
-  }
-  def trace(e: => Throwable): Any = {
-    if (logger.isTraceEnabled())
-      logger.trace(logIdent,e)
+  val instance = Try {
+    val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
+    val instance = log4jController.newInstance()
```

Comment:
`c.getDeclaredConstructor().newInstance()` is preferred as of Java 9.

Reply:
I'll change that. Thanks for the heads up.